--- Makefile    Tue Apr 23 15:13:01 2002
+++ q2dedMakefile       Thu Apr 25 00:54:10 2002
@@ -100,6 +100,7 @@
 CTF_DIR=$(MOUNT_DIR)/ctf
 XATRIX_DIR=$(MOUNT_DIR)/xatrix
 ROGUE_DIR=$(MOUNT_DIR)/rogue
+NULL_DIR=$(MOUNT_DIR)/null

 BASE_CFLAGS=-Wall -pipe -Dstricmp=strcasecmp
 ifeq ($(HAVE_IPV6),YES)
@@ -160,6 +161,8 @@
 SHLIBLDFLAGS=-shared

 DO_CC=$(CC) $(CFLAGS) -o $@ -c $<
+DO_DED_CC=$(CC) $(CFLAGS) -DDEDICATED_ONLY -o $@ -c $<
+DO_DED_DEBUG_CC=$(CC) $(DEBUG_CFLAGS) -DDEDICATED_ONLY -o $@ -c $<
 DO_SHLIB_CC=$(CC) $(CFLAGS) $(SHLIBCFLAGS) -o $@ -c $<
 DO_GL_SHLIB_CC=$(CC) $(CFLAGS) $(SHLIBCFLAGS) $(GLCFLAGS) -o $@ -c $<
 DO_AS=$(CC) $(CFLAGS) -DELF -x assembler-with-cpp -o $@ -c $<
@@ -171,7 +174,7 @@

 .PHONY : targets build_debug build_release clean clean-debug clean-release clean2

-TARGETS=$(BUILDDIR)/quake2 $(BUILDDIR)/game$(ARCH).$(SHLIBEXT)
+TARGETS=$(BUILDDIR)/quake2 $(BUILDDIR)/q2ded $(BUILDDIR)/game$(ARCH).$(SHLIBEXT)

 ifeq ($(strip $(BUILD_CTFDLL)),YES)
  TARGETS += $(BUILDDIR)/ctf/game$(ARCH).$(SHLIBEXT)
@@ -310,6 +313,7 @@
 build_debug:
        @-mkdir -p $(BUILD_DEBUG_DIR) \
                $(BUILD_DEBUG_DIR)/client \
+               $(BUILD_DEBUG_DIR)/ded \
                $(BUILD_DEBUG_DIR)/ref_soft \
                $(BUILD_DEBUG_DIR)/ref_gl \
                $(BUILD_DEBUG_DIR)/game \
@@ -321,6 +325,7 @@
 build_release:
        @-mkdir -p $(BUILD_RELEASE_DIR) \
                $(BUILD_RELEASE_DIR)/client \
+               $(BUILD_RELEASE_DIR)/ded \
                $(BUILD_RELEASE_DIR)/ref_soft \
                $(BUILD_RELEASE_DIR)/ref_gl \
                $(BUILD_RELEASE_DIR)/game \
@@ -564,6 +569,116 @@
        $(DO_CC) $(SDLCFLAGS)

 #############################################################################
+# DEDICATED SERVER
+#############################################################################
+
+Q2DED_OBJS = \
+       \
+       $(BUILDDIR)/ded/cmd.o \
+       $(BUILDDIR)/ded/cmodel.o \
+       $(BUILDDIR)/ded/common.o \
+       $(BUILDDIR)/ded/crc.o \
+       $(BUILDDIR)/ded/cvar.o \
+       $(BUILDDIR)/ded/files.o \
+       $(BUILDDIR)/ded/md4.o \
+       $(BUILDDIR)/ded/net_chan.o \
+       \
+       $(BUILDDIR)/ded/sv_ccmds.o \
+       $(BUILDDIR)/ded/sv_ents.o \
+       $(BUILDDIR)/ded/sv_game.o \
+       $(BUILDDIR)/ded/sv_init.o \
+       $(BUILDDIR)/ded/sv_main.o \
+       $(BUILDDIR)/ded/sv_send.o \
+       $(BUILDDIR)/ded/sv_user.o \
+       $(BUILDDIR)/ded/sv_world.o \
+       \
+       $(BUILDDIR)/ded/q_shlinux.o \
+       $(BUILDDIR)/ded/sys_linux.o \
+       $(BUILDDIR)/ded/glob.o \
+       $(BUILDDIR)/ded/net_udp.o \
+       \
+       $(BUILDDIR)/ded/q_shared.o \
+       $(BUILDDIR)/ded/pmove.o \
+       \
+       $(BUILDDIR)/ded/cl_null.o \
+       $(BUILDDIR)/ded/cd_null.o
+
+$(BUILDDIR)/q2ded : $(Q2DED_OBJS)
+       $(CC) $(CFLAGS) -o $@ $(Q2DED_OBJS) $(LDFLAGS)
+
+$(BUILDDIR)/ded/cmd.o :        $(COMMON_DIR)/cmd.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/cmodel.o :     $(COMMON_DIR)/cmodel.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/common.o :     $(COMMON_DIR)/common.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/crc.o :        $(COMMON_DIR)/crc.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/cvar.o :       $(COMMON_DIR)/cvar.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/files.o :      $(COMMON_DIR)/files.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/md4.o :        $(COMMON_DIR)/md4.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/net_chan.o :   $(COMMON_DIR)/net_chan.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/q_shared.o :   $(GAME_DIR)/q_shared.c
+       $(DO_DED_DEBUG_CC)
+
+$(BUILDDIR)/ded/pmove.o :      $(COMMON_DIR)/pmove.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/sv_ccmds.o :   $(SERVER_DIR)/sv_ccmds.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/sv_ents.o :    $(SERVER_DIR)/sv_ents.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/sv_game.o :    $(SERVER_DIR)/sv_game.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/sv_init.o :    $(SERVER_DIR)/sv_init.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/sv_main.o :    $(SERVER_DIR)/sv_main.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/sv_send.o :    $(SERVER_DIR)/sv_send.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/sv_user.o :    $(SERVER_DIR)/sv_user.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/sv_world.o :   $(SERVER_DIR)/sv_world.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/q_shlinux.o :  $(LINUX_DIR)/q_shlinux.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/sys_linux.o :  $(LINUX_DIR)/sys_linux.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/glob.o :       $(LINUX_DIR)/glob.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/net_udp.o :    $(LINUX_DIR)/net_udp.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/cd_null.o :    $(NULL_DIR)/cd_null.c
+       $(DO_DED_CC)
+
+$(BUILDDIR)/ded/cl_null.o :    $(NULL_DIR)/cl_null.c
+       $(DO_DED_CC)
+
+#############################################################################
 # GAME
 #############################################################################

@@ -1161,6 +1276,106 @@
        $(BUILDDIR)/rogue/p_view.o \
        $(BUILDDIR)/rogue/p_weapon.o \
        $(BUILDDIR)/rogue/q_shared.o
+
+$(BUILDDIR)/rogue/game$(ARCH).$(SHLIBEXT) : $(ROGUE_OBJS)
+       $(CC) $(CFLAGS) $(SHLIBLDFLAGS) -o $@ $(ROGUE_OBJS)
+
+$(BUILDDIR)/rogue/dm_ball.o :      $(ROGUE_DIR)/dm_ball.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/dm_tag.o :       $(ROGUE_DIR)/dm_tag.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_ai.o :         $(ROGUE_DIR)/g_ai.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_chase.o :      $(ROGUE_DIR)/g_chase.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_cmds.o :       $(ROGUE_DIR)/g_cmds.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_combat.o :     $(ROGUE_DIR)/g_combat.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_func.o :       $(ROGUE_DIR)/g_func.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_items.o :      $(ROGUE_DIR)/g_items.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_main.o :       $(ROGUE_DIR)/g_main.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_misc.o :       $(ROGUE_DIR)/g_misc.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_monster.o :    $(ROGUE_DIR)/g_monster.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_newai.o :      $(ROGUE_DIR)/g_newai.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_newdm.o :      $(ROGUE_DIR)/g_newdm.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_newfnc.o :     $(ROGUE_DIR)/g_newfnc.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_newtarg.o :    $(ROGUE_DIR)/g_newtarg.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_newtrig.o :    $(ROGUE_DIR)/g_newtrig.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_newweap.o :    $(ROGUE_DIR)/g_newweap.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_phys.o :       $(ROGUE_DIR)/g_phys.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_save.o :       $(ROGUE_DIR)/g_save.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_spawn.o :      $(ROGUE_DIR)/g_spawn.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_sphere.o :     $(ROGUE_DIR)/g_sphere.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_svcmds.o :     $(ROGUE_DIR)/g_svcmds.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_target.o :     $(ROGUE_DIR)/g_target.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_trigger.o :    $(ROGUE_DIR)/g_trigger.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_turret.o :     $(ROGUE_DIR)/g_turret.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_utils.o :      $(ROGUE_DIR)/g_utils.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/g_weapon.o :     $(ROGUE_DIR)/g_weapon.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/m_actor.o :      $(ROGUE_DIR)/m_actor.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/m_berserk.o :    $(ROGUE_DIR)/m_berserk.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/m_boss2.o :      $(ROGUE_DIR)/m_boss2.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/m_boss3.o :      $(ROGUE_DIR)/m_boss3.c
+       $(DO_SHLIB_CC)
+
+$(BUILDDIR)/rogue/m_boss31.o :     $(ROGUE_DIR)/m_boss31.c
+       $(DO_SHLIB_CC)
+
 $(BUILDDIR)/rogue/m_boss32.o :     $(ROGUE_DIR)/m_boss32.c
        $(DO_SHLIB_CC)

@@ -1532,6 +1747,7 @@
        $(QUAKE2_AS_OBJS) \
        $(GAME_OBJS) \
        $(CTF_OBJS) \
+       $(ROGUE_OBJS) \
        $(XATRIX_OBJS) \
        $(REF_SOFT_OBJS) \
        $(REF_SOFT_SVGA_OBJS) \
@@ -1542,3 +1758,18 @@
        @-rm -rf $(BUILD_DEBUG_DIR) $(BUILD_RELEASE_DIR)
        @-rm -f `find . \( -not -type d \) -and \
                \( -name '*~' \) -type f -print`
+
